contracts/MyMultiSig.cdc

```diff
@@ -23,20 +12,68 @@ pub contract MyMultiSigV3 {
         access(contract) let multiSignManager: @Manager
     }
 
+    pub resource interface ManagerPublic {
+        pub fun borrowAction(actionUUID: UInt64): &MultiSignAction
+        pub fun getIDs(): [UInt64]
+        pub fun getIntents(): {UInt64: String}
+        pub fun getSigners(): {Address: Bool}
+        pub fun getThreshold(): UInt
+        pub fun getSignerResponsesForAction(actionUUID: UInt64): {Address: UInt}
+        pub fun getTotalApprovedForAction(actionUUID: UInt64): Int
+        pub fun getTotalRejectedForAction(actionUUID: UInt64): Int
+    }
+
     //
-    // ------- Action Wrapper ------- 
+    // ------- Struct Interfaces ------- 
     //
 
+    // Interface for actions implemented in TreasuryActions contract
     pub struct interface Action {
         pub let intent: String
         pub let proposer: Address
+        pub fun getView(): ActionView
         access(account) fun execute(_ params: {String: AnyStruct})
     }
 
     //
-    // ------- Structs --------
+    // ------- Structs ------- 
     //
 
+    // General View Struct for Actions
+    pub struct ActionView {
+        pub var name: String
+        pub var type: String
```

Comment:
Can we in cadence define a enum of all action types and define the type here to be one of those enums?

Reply:
we can definitely do it, but i'm not sure it will actually provide many benefits.  this is because an enum ultimately maps values to a `UInt`, so we would either need to store our own map of `UInt -> String` in the script file, or in our FE javascript, so in my opinion it is more trouble than it is worth.

is there a benefit you think we are missing out on if we don't store these as Enums?